flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectResultFetcher.java

```diff
@@ -125,7 +127,13 @@ public T next() throws IOException {
                 try {
                     response = sendRequest(buffer.getVersion(), requestOffset);
                 } catch (Exception e) {
-                    LOG.warn("An exception occurs when fetching query results", e);
+                    if (ExceptionUtils.findThrowable(
+                                    e, UnavailableDispatcherOperationException.class)
+                            .isPresent()) {
+                        LOG.debug("The job execution has not started yet; cannot fetch results.");
```

Comment:
It's difficult to solve, yes. I think we just need to be more conscious about changing what exceptions we throw; I would hope that in this case, because it is a fairly specific exception, that we don't have to worry as much.